feat(api-client): add Postman import

Mason-Little opened the pull request:
**Overview**

Added native support for importing Postman collections through our standard import modal. Users can now seamlessly import Postman collections as if they were OpenAPI specs, with automatic detection and conversion.

- Added Postman collection detection and conversion
- Created postman.ts for handling Postman-specific functionality
- Enhanced import modal to handle both OpenAPI and Postman specs transparently
- Integrated @scalar/postman-to-openapi for reliable conversions

hanspagel commented:
* lock file got a quick update, no deletes, we chill
* tightened up some code style, looks clean now
* tossed in a changeset

hmu if it’s sus or anything’s off

hanspagel commented on packages/api-client/src/libs/postman.ts:
```diff
@@ -0,0 +1,46 @@
+    const parsed = JSON.parse(content)
+    const isPostman =
+      parsed.info?._postman_id !== undefined &&
+      parsed.info?.schema?.includes('schema.getpostman.com')
```
✅ done

amritk reviewed (approved):
looks gooooooood

antlio commented:
looks good! should we update the wording in the import modal accordingly?

marclave reviewed (changes requested):
i converted petstore openapi -> postman collection then import but i dont have any servers

<img width="766" alt="image" src="https://github.com/user-attachments/assets/b146a56c-265e-47cf-88eb-85fb12616ef2">

<img width="1293" alt="image" src="https://github.com/user-attachments/assets/d95bc01f-77af-48e6-b7c0-4cee71b30e93">

marclave commented:
i converted Scalar Galaxy -> Postman collection then imported it and it fails

[Scalar Galaxy.postman_collection.json](https://github.com/user-attachments/files/18021573/Scalar.Galaxy.postman_collection.json)

marclave commented:
@Mason-Little i updated the copy to make it searchable : )

<img width="624" alt="image" src="https://github.com/user-attachments/assets/6a47ee6b-04d9-4b55-a13f-1299df8ee933">

<img width="312" alt="image" src="https://github.com/user-attachments/assets/8e46426a-df5b-45b3-8915-30ee09618ff7">

<img width="821" alt="image" src="https://github.com/user-attachments/assets/8f351b15-e2e7-4a9e-bfc9-d15eff9b5fc5">

marclave commented:
> looks good! should we update the wording in the import modal accordingly?

done :)

Mason-Little commented:
> i converted Scalar Galaxy -> Postman collection then imported it and it fails
> 
> [Scalar Galaxy.postman_collection.json](https://github.com/user-attachments/files/18021573/Scalar.Galaxy.postman_collection.json)

fixed with noAuth added in `authHelpers.ts`

Mason-Little commented:
> i converted petstore openapi -> postman collection then import but i dont have any servers
> 
> <img alt="image" width="766" src="https://private-user-images.githubusercontent.com/6176314/392769091-b146a56c-265e-47cf-88eb-85fb12616ef2.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzM0MDkwOTIsIm5iZiI6MTczMzQwODc5MiwicGF0aCI6Ii82MTc2MzE0LzM5Mjc2OTA5MS1iMTQ2YTU2Yy0yNjVlLTQ3Y2YtODhlYi04NWZiMTI2MTZlZjIucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MTIwNSUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDEyMDVUMTQyNjMyWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9NzY3MWRlYzMxYjdjYzk0ZjE0YjRmYzVhMzIwMWUzMTk3NDc4YmMyMjIxYjU3NDIxOGUyNTBkYzhhNDFiMzRjYiZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.ysE81qGrfn_EiSLWldbS6aqVph-WmVmyZyqKeqrK51g"> <img alt="image" width="1293" src="https://private-user-images.githubusercontent.com/6176314/392769152-d95bc01f-77af-48e6-b7c0-4cee71b30e93.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzM0MDkwOTIsIm5iZiI6MTczMzQwODc5MiwicGF0aCI6Ii82MTc2MzE0LzM5Mjc2OTE1Mi1kOTViYzAxZi03N2FmLTQ4ZTYtYjdjMC00Y2VlNzFiMzBlOTMucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MTIwNSUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDEyMDVUMTQyNjMyWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9ZTMyNTA4M2E4ZWVhOGZhZjI3NTU0OTcxZjFkNjc1MGRlYmIyMTUxM2IwOTkzMzExYzVhYThkMzk4MWU4YmM3ZiZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.nLzTNKrj3gizkW10ixtVvAm1EqDZd1KncsTZFASpV5w">
> [Swagger Petstore - OpenAPI 3.0.postman_collection.json](https://github.com/user-attachments/files/18021567/Swagger.Petstore.-.OpenAPI.3.0.postman_collection.json)

fixed with [updated Postman-to-openapi](https://github.com/scalar/scalar/pull/4091)
